util/OptionsDB.h

```diff
@@ -338,8 +344,23 @@ class FO_COMMON_API OptionsDB
 
     std::map<std::string, Option>   m_options;
     static OptionsDB*               s_options_db;
+    bool                            m_dirty; //< has OptionsDB changed since last Commit()
 
     friend OptionsDB& GetOptionsDB();
 };
 
+template <typename T>
+bool OptionsDB::Option::SetFromValue(const T& value_) {
+    if (value.type() != typeid(T))
```

Comment:
I can change this but I'd like to keep it:
1. It was moved from OptionsDB.cpp:290, so it has been in the code base for some time.
2. It is the standard idiom for boost::any, right on the front page example.  (The very next example on the same page is try{} catch(boost::bad_any_cast &) )
3. It is less verbose.